packages/protobuf/src/private/binary-format-common.ts

```diff
@@ -91,14 +90,22 @@ export function makeBinaryFormatCommon(): Omit<BinaryFormat, "writeMessage"> {
     readMessage<T extends Message<T>>(
       message: T,
       reader: IBinaryReader,
-      length: number,
+      lengthOrEndTagFieldNo: number,
       options: BinaryReadOptions,
+      delimitedMessageEncoding?: boolean,
     ): void {
       const type = message.getType();
-      const end = length === undefined ? reader.len : reader.pos + length;
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+      const end = delimitedMessageEncoding
+        ? reader.len
+        : reader.pos + lengthOrEndTagFieldNo;
```

Comment:
This is a semantic change right? Previously, an `undefined` length could be provided, but this new expression doesn't look like it will handle that the same way/correctly. Should the first part of the ternery instead be `delimitedMessageEncoding || length === undefined`?

Reply:
This threw me off as well. The code used to handle length being `undefined`, but the signature does not allow `undefined` to be passed, and I couldn't find any indication it ever did. I think this must have been left over from initial development.